### PR TITLE
[MM-20568] Migrate 'components/admin_console/system_users' module to ts

### PR DESCRIPTION
**Pull request by vraravam**
#### Summary
Converted 'system_users' module to ts

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13504

#### Related Pull Requests
This PR depends on https://github.com/mattermost/mattermost-webapp/pull/6759 to be merged prior (I can resolve conflicts once that is done)

### CODE REVIEW

File: components/admin_console/system_users/system_users.tsx
Comment (jespino) [inline components/admin_console/system_users/system_users.tsx:34]: Shouldn't it be `Team[]`?

File: components/admin_console/system_users/system_users.tsx
Comment (jespino) [inline components/admin_console/system_users/system_users.tsx:59]: I think this could be `Record<string, UserProfile>`

File: components/admin_console/system_users/system_users.tsx
Comment (jespino) [inline components/admin_console/system_users/system_users.tsx:246]: This should be `string`

**Review by jespino — CHANGES_REQUESTED** [review]
Some improvement recommendations on types, but is looking good for me.

File: components/admin_console/system_users/list/system_users_list.jsx
Comment (jespino) [inline components/admin_console/system_users/list/system_users_list.jsx:24]: Nice catches here with the missing fields :)

Comment (vraravam) [reply components/admin_console/system_users/list/system_users_list.jsx:24]: Thanks @jespino - have fixed this in the latest commit.

File: components/admin_console/system_users/system_users.tsx
Comment (vraravam) [reply components/admin_console/system_users/system_users.tsx]: Thanks @jespino - have fixed this in the latest commit.

File: components/admin_console/system_users/system_users.tsx
Comment (vraravam) [reply components/admin_console/system_users/system_users.tsx]: Thanks @jespino - have fixed this in the latest commit.

File: components/admin_console/system_users/system_users.tsx
Comment (vraravam) [reply components/admin_console/system_users/system_users.tsx]: Thanks @jespino - have fixed this in the latest commit.

**Comment by vraravam** [comment]
> Some improvement recommendations on types, but is looking good for me.

@jespino  @devinbinnie - can you please review the latest again ? I have rebased from the `master` branch and would like some help to get this merged asap if there are no other review comments.

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @vraravam! :tada:

**Comment by vraravam** [comment]
> Looks good to me. Thanks @vraravam! 🎉

The tag "Do not Merge/Awaiting PR" label can be removed - the PR that was needed has been merged a few days ago - and I have rebased from the `master` branch.

**Comment by vraravam** [comment]
@lieut-data / @ogi-m - any idea when you would be able to take a look and approve this PR? Its holding up at least 1 other PR that I am working on.

**Review by lieut-data — APPROVED** [review]
Sorry for the delay! Was out of office -- thanks for the changes :)

**Comment by vraravam** [comment]
thanks @lieut-data - whats next to get this PR merged?

**Comment by jasonblais** [comment]
@vraravam thanks for the contribution! There is one more QA review by @ogi-m remaining after which the PR can be merged.

You can learn more about our code review process at https://developers.mattermost.com/contribute/getting-started/code-review/. Code changes are reviewed by at least two core committers (which is completed) and a QA tester (remaining).

**Review by ogi-m — APPROVED** [review]
Smoke tested the system users page and everything's working as expected.

LGTM 👍

**Comment by ashishbhate** [comment]
Thanks @vraravam!

**Pull request merged**
